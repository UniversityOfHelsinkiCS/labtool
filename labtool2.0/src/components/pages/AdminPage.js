import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Button, Loader } from 'semantic-ui-react'
import { Redirect } from 'react-router'
import { getAllCI } from '../../services/courseInstance'
import { resetLoading } from '../../reducers/loadingReducer'
import { getAllUsers, updateOtherUser } from '../../services/user'
import { clearNotifications } from '../../reducers/notificationReducer'
import { usePersistedState } from '../../hooks/persistedState'
import { AdminCourseList } from './AdminPage/AdminCourseList'
import { AdminUserList } from './AdminPage/AdminUserList'
import DocumentTitle from '../DocumentTitle'
import Error from '../Error'

export const AdminPage = props => {
  const state = usePersistedState('AdminPage', {
    coursesList: false,
    usersList: false
  })

  useEffect(() => {
    props.clearNotifications()
  }, [])

  useEffect(() => {
    if (state.coursesList) {
      props.resetLoading()
      props.getAllCI()
    }
  }, [state.coursesList])

  useEffect(() => {
    if (state.usersList) {
      props.resetLoading()
      props.getAllUsers()
    }
  }, [state.usersList])

  const documentTitle = <DocumentTitle title={'Admin'} />

  if (!props.user.user.sysop) {
    return <Redirect to={`/labtool`} />
  } else if (props.errors && props.errors.length > 0) {
    return <Error errors={props.errors.map(error => `${error.response.data} (${error.response.status} ${error.response.statusText})`)} />
  } else if (props.loading.loading) {
    return (
      <>
        <Loader active />
        {documentTitle}
      </>
    )
  }

  return (
    <>
      {documentTitle}
      <div className="AdminPage">
        <h2>Welcome to the admin interface.</h2>
        <Button compact className={`ui button`} onClick={() => (state.coursesList = !state.coursesList)}>
          {state.coursesList ? 'Hide courses' : 'Manage courses'}
        </Button>
        <br />
        <br />
        {state.coursesList && <AdminCourseList courses={props.courseInstance} />}
        <br />
        <hr />
        <br />
        <Button compact className={`ui button`} onClick={() => (state.usersList = !state.usersList)}>
          {state.usersList ? 'Hide users' : 'Manage users'}
        </Button>
        <br />
        {state.usersList && <AdminUserList me={props.user} users={props.users} updateOtherUser={props.updateOtherUser} />}
      </div>
    </>
  )
}

const mapStateToProps = state => {
  return {
    user: state.user,
    courseInstance: state.courseInstance,
    users: state.users,
    loading: state.loading,
    errors: Object.values(state.loading.errors)
  }
}

const mapDispatchToProps = {
  getAllCI,
  resetLoading,
  getAllUsers,
  updateOtherUser,
  clearNotifications
}

AdminPage.propTypes = {
  user: PropTypes.object.isRequired,
  courseInstance: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  users: PropTypes.array.isRequired,
  loading: PropTypes.object.isRequired,

  getAllCI: PropTypes.func.isRequired,
  resetLoading: PropTypes.func.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  updateOtherUser: PropTypes.func.isRequired,
  clearNotifications: PropTypes.func.isRequired,

  errors: PropTypes.array
}

export default connect(mapStateToProps, mapDispatchToProps)(AdminPage)
