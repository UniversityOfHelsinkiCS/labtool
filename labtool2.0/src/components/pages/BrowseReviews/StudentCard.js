import React from 'react'
import PropTypes from 'prop-types'
import { Button, Card, Icon, Label, Header, Popup } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { createCourseIdWithYearAndTerm } from '../../../util/format'

import RepoLink from '../../RepoLink'

export const StudentCard = ({ student, otherParticipations, handleMarkAsDropped, handleMarkAsValidRegistration, teacherInstance }) => (
  <Card key={student.id} fluid color="yellow" className="studentCard">
    <Card.Content>
      <Header as="h2">
        {student.User.firsts} {student.User.lastname} ({student.User.studentNumber})
        {student.dropped && (
          <Label style={{ marginTop: 5, float: 'right' }}>
            <Icon name="warning sign" color="red" />
            Has dropped course
          </Label>
        )}
        {!student.validRegistration && (
          <Label style={{ marginTop: 5, float: 'right' }}>
            <Icon name="warning" color="red" />
            Invalid course registration
          </Label>
        )}
        <Header.Subheader>
          <a href={`mailto:${student.User.email}`}>{student.User.email}</a>
        </Header.Subheader>
      </Header>
      <div style={{ display: 'inline-block' }}>
        {student.projectName}: <RepoLink url={student.github} />
        <br />
        {otherParticipations.length > 0 ? (
          <div className="hasOther">
            <p style={{ color: 'red' }}>Has taken this course in other periods</p>
            {otherParticipations.map(participation =>
              teacherInstance.find(course => course.id === participation.id) ? (
                //show link to participation if the teacher/assistant has been the instructor in that course, otherwise only text
                <Link key={participation.id} to={`/labtool/browsereviews/${participation.ohid}/${participation.courseInstances[0].id}`}>
                  <Popup content="click to see the details" trigger={<p>{createCourseIdWithYearAndTerm(participation.ohid, participation.start)}</p>} />
                </Link>
              ) : (
                <p className="noAccess" key={participation.id}>
                  {createCourseIdWithYearAndTerm(participation.ohid, participation.start)}
                </p>
              )
            )}
          </div>
        ) : (
          <p className="noOther">Has no other participation in this course</p>
        )}
      </div>
      {
        <Button color="red" style={{ float: 'right' }} onClick={() => handleMarkAsDropped(!student.dropped)}>
          {student.dropped ? 'Mark as non-dropped' : 'Mark as dropped'}
        </Button>
      }
      {
        <Popup
          content="Mark a registration as invalid, if the student accidentally registered onto the course"
          trigger={
            <Button basic id="buttonInvalidRegistration" color="red" style={{ float: 'right' }} onClick={() => handleMarkAsValidRegistration(!student.validRegistration)}>
              {student.validRegistration ? 'Mark registration as invalid' : 'Mark registration as valid'}
            </Button>
          }
        />
      }
    </Card.Content>
  </Card>
)

StudentCard.propTypes = {
  student: PropTypes.object.isRequired,
  teacherInstance: PropTypes.array.isRequired,
  otherParticipations: PropTypes.array.isRequired,
  handleMarkAsDropped: PropTypes.func.isRequired,
  handleMarkAsValidRegistration: PropTypes.func.isRequired
}

export default StudentCard
