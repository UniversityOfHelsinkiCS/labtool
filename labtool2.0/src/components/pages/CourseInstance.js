import React from 'react'

const CourseInstance = ({ instance, handleFieldChange }) => {
  return(
    <li> {instance.name} <button onClick={handleFieldChange} name="courseInstanceId" value={instance.id}>Register</button>
    </li>
  )
}

export default CourseInstance