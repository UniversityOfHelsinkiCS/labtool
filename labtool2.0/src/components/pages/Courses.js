import React, { Component } from 'react'
import { Button, List, Container, Header, Table, Label } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

import { getAllCI } from '../../services/courseInstance'

/**
 *  Show all the courses in a single list.
 */
class Courses extends Component {
  componentWillMount() {
    this.props.getAllCI()
  }

  render() {
    return (
      <div>
        <Container>
          <Header as="h2">Courses</Header>
          <Table singleLine color="yellow">
            <Table.Header>
              <Table.Row>
                <Table.HeaderCell colSpan="1"> </Table.HeaderCell>
                <Table.HeaderCell colSpan="1">Course id</Table.HeaderCell>
                <Table.HeaderCell colSpan="1">Course name</Table.HeaderCell>
                <Table.HeaderCell colSpan="1">Course start date</Table.HeaderCell>
                <Table.HeaderCell colSpan="2"> </Table.HeaderCell>
              </Table.Row>
            </Table.Header>

            <Table.Body>
              {this.props.courseInstance.map(instance => (
                <Table.Row key={instance.id}>
                  <Table.Cell>
                    <div>
                      {instance.active === true ? (
                        <Label ribbon style={{ backgroundColor: '#21ba45' }}>
                          Active
                        </Label>
                      ) : (
                          ''
                        )}
                    </div>
                  </Table.Cell>
                  <Table.Cell>{instance.shorterId} </Table.Cell>
                  <Table.Cell>
                    <strong>
                      <a href={`/labtool/courses/${instance.ohid}`}>{instance.name}</a>
                    </strong>
                  </Table.Cell>

                  <Table.Cell> {instance.europeanStart} </Table.Cell>
                  <Table.Cell textAlign="center">
                    <Button circular size="tiny" icon={{ name: 'eye', size: 'large', color: 'blue' }} as={Link} to={`/labtool/courses/${instance.ohid}`} />
                  </Table.Cell>
                </Table.Row>
              ))}
            </Table.Body>
          </Table>

          <div className="Instructions">
            <List>
              <List.Item icon={{ name: 'eye', color: 'blue' }} content="Show course page" />
              <List.Item icon={{ name: 'square', color: 'green' }} content="Course is activated" />
            </List>
          </div>
        </Container>
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    courseInstance: state.courseInstance
  }
}

export default connect(mapStateToProps, { getAllCI })(Courses)
