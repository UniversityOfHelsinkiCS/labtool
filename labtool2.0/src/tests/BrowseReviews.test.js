import React from 'react'
import { BrowseReviews } from '../components/pages/BrowseReviews'
import { shallow } from 'enzyme'

describe('<BrowseReviews />', () => {
  let wrapper

  const currentCourseId = 10011
  const otherParticipationId = 10000

  const coursePage = {
    id: 10011,
    name: 'Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit (periodi IV)',
    start: '2018-03-11T21:00:00.000Z',
    end: '2018-04-29T21:00:00.000Z',
    active: true,
    weekAmount: 7,
    weekMaxPoints: 3,
    currentWeek: 1,
    ohid: 'TKT20010.2018.K.A.1',
    teacherInstances: [
      {
        id: 10001,
        instructor: false,
        userId: 10010,
        courseInstanceId: 10011,
        firsts: 'Pää',
        lastname: 'Opettaja'
      },
      {
        id: 10011,
        instructor: true,
        userId: 10015,
        courseInstanceId: 10011,
        firsts: 'Ossi Ohjaaja',
        lastname: 'Mutikainen'
      }
    ]
  }

  const courseData = {
    role: 'teacher',
    data: [
      {
        id: 10012,
        github: 'http://github.com/tiralabra2',
        projectName: 'Tiran toinen labraprojekti',
        dropped: false,
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-03-26T00:00:00.000Z',
        courseInstanceId: 10011,
        userId: 10012,
        teacherInstanceId: 10011,
        weeks: [],
        User: {
          id: 10012,
          username: 'tiraopiskelija2',
          email: 'johan.studerande@helsinki.invalid',
          firsts: 'Johan Wilhelm',
          lastname: 'Studerande',
          studentNumber: '014553242',
          teacher: false,
          sysop: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        }
      },
      {
        id: 10031,
        github: 'http://github.com/superprojekti',
        projectName: 'Tira super projekti',
        dropped: false,
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-06-05T07:12:28.603Z',
        courseInstanceId: 10011,
        userId: 10031,
        teacherInstanceId: 10011,
        weeks: [],
        User: {
          id: 10031,
          username: 'superopiskelija',
          email: 'teras.henkilo@helsinki.invalid',
          firsts: 'Teräs',
          lastname: 'Henkilö',
          studentNumber: '014666666',
          teacher: false,
          sysop: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        }
      },
      {
        id: 10011,
        github: 'http://github.com/tiralabra1',
        projectName: 'Tiran labraprojekti',
        dropped: false,
        createdAt: '2018-03-26T00:00:00.000Z',
        updatedAt: '2018-03-26T00:00:00.000Z',
        courseInstanceId: 10011,
        userId: 10011,
        teacherInstanceId: 10011,
        weeks: [
          {
            id: 10002,
            points: 2,
            weekNumber: 2,
            feedback: 'Melko hienoa työtä!',
            instructorNotes: 'READMEssa ongelmia',
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z',
            studentInstanceId: 10011,
            comments: []
          },
          {
            id: 10001,
            points: 3,
            weekNumber: 1,
            feedback: 'Hienoa työtä!',
            instructorNotes: '',
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z',
            studentInstanceId: 10011,
            comments: []
          },
          {
            id: 10003,
            points: 3,
            weekNumber: 3,
            feedback: 'Erittäin hienoa työtä!',
            instructorNotes: '',
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z',
            studentInstanceId: 10011,
            comments: []
          },
          {
            id: 10004,
            points: 3,
            weekNumber: 4,
            feedback: 'Hyvin menee!',
            instructorNotes: '',
            createdAt: '2018-03-26T00:00:00.000Z',
            updatedAt: '2018-03-26T00:00:00.000Z',
            studentInstanceId: 10011,
            comments: []
          }
        ],
        codeReviews: [
          {
            id: 272,
            points: null,
            revieweNumber: 1,
            linkToReview: 'https:github.com/toReview/repo/pull/1',
            repoToReview: null,
            studentInstanceId: 10011,
            toReview: 10031
          },
          {
            d: 273,
            points: null,
            revieweNumber: 2,
            linkToReview: null,
            repoToReview: 'https://github.com/userName/arbitraryRepo',
            studentInstanceId: 10011,
            toReview: null
          }
        ],
        User: {
          id: 10011,
          username: 'tiraopiskelija1',
          email: 'maarit.opiskelija@helsinki.invalid',
          firsts: 'Maarit Mirja',
          lastname: 'Opiskelija',
          studentNumber: '014578343',
          teacher: false,
          sysop: false,
          createdAt: '2018-03-26T00:00:00.000Z',
          updatedAt: '2018-03-26T00:00:00.000Z'
        }
      }
    ]
  }

  const loading = {
    loading: false,
    loadingHooks: [],
    redirect: false,
    redirectHooks: [],
    redirectFailure: false
  }

  let mockFn = jest.fn()

  let mockUpdateStudentProjectInfo

  const studentWithoutPreviousParticipation = [coursePage]

  const studentInstanceId = '10011'

  const teacherCourses = [
    {
      id: otherParticipationId
    },
    {
      id: currentCourseId
    }
  ]

  beforeEach(() => {
    mockUpdateStudentProjectInfo = jest.fn()

    wrapper = shallow(
      <BrowseReviews
        getOneCI={mockFn}
        coursePageInformation={mockFn}
        getCoursesByStudentId={mockFn}
        updateStudentProjectInfo={mockUpdateStudentProjectInfo}
        courseData={courseData}
        selectedInstance={coursePage}
        studentInstanceToBeReviewed={studentWithoutPreviousParticipation}
        teacherInstance={teacherCourses}
        getAllTeacherCourses={mockFn}
        courseId={coursePage.ohid}
        studentInstance={studentInstanceId} //studentInstance id which was chosen randomly from courseData
        loading={loading}
        resetLoading={mockFn}
        initialLoading={false}
        user={{}}
        createOneComment={mockFn}
        gradeCodeReview={mockFn}
        sendEmail={mockFn}
      />
    )
  })

  describe('BrowseReviews Component', () => {
    it('is ok', () => {
      true
    })

    it('should render without throwing an error', () => {
      expect(wrapper.find('.BrowseReviews').exists()).toEqual(true)
    })

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    describe('student card', () => {
      it('should render student card', () => {
        expect(wrapper.find('StudentCard').exists()).toEqual(true)
      })
      it('when student participates the course first time', () => {
        wrapper = wrapper.find('StudentCard').dive()
        expect(wrapper.find('.noOther').text()).toEqual('Has no other participation in this course')
      })
      describe('when student participate other instances of the course', () => {
        const studentWithOtherParticipation = [
          {
            ...coursePage,
            id: otherParticipationId,
            name: 'Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit',
            start: '2017-03-11T21:00:00.000Z',
            end: '2017-04-29T21:00:00.000Z',
            active: false,
            ohid: 'TKT20010.2017.K.A.1',
            courseInstances: [{ id: 1, validRegistration: true }]
          },
          {
            ...coursePage,
            courseInstances: [{ id: 10011, validRegistration: true }]
          }
        ]
        beforeEach(() => {
          wrapper.setProps({ studentInstanceToBeReviewed: studentWithOtherParticipation, studentInstanceId: '10011' })
          wrapper = wrapper.find('StudentCard').dive()
        })

        it('Teacher has access to other participation', () => {
          expect(wrapper.find('.hasOther').text()).toContain('Has taken this course in other periods')
          const popup = wrapper
            .find('.hasOther')
            .find('Link')
            .find('Popup')
          expect(popup.props()).toHaveProperty('trigger', <p>TKT20010 2016-2017 P.IV</p>)
        })

        it('Teacher has no access to other participation', () => {
          wrapper.setProps({ teacherInstance: [{ id: currentCourseId }] })
          expect(
            wrapper
              .find('.hasOther')
              .find('Link')
              .exists()
          ).toBeFalsy()
          expect(
            wrapper
              .find('.hasOther')
              .find('.noAccess')
              .text()
          ).toContain('TKT20010 2016-2017 P.IV')
        })
      })
      it('student can be marked as dropped and non-dropped', () => {
        wrapper = wrapper.find('StudentCard').dive()
        wrapper.find({ children: 'Mark as dropped' }).simulate('click')

        expect(mockUpdateStudentProjectInfo).toBeCalledWith(expect.objectContaining({ dropped: true }))
      })
      it('students course registration can be marked as valid and invalid', () => {
        wrapper = wrapper.find('StudentCard').dive()
        //wrapper.find({ children: 'Mark registration as valid' }).simulate('click')
        shallow(wrapper.find('Popup').prop('trigger')).dive().find('#buttonInvalidRegistration').simulate('click')

        expect(mockUpdateStudentProjectInfo).toBeCalledWith(expect.objectContaining({ validRegistration: true }))
      })
    })
  })
})
