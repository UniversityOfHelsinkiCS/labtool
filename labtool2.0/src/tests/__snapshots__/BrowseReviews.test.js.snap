// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowseReviews /> BrowseReviews Component should render correctly 1`] = `
<div
  className="BrowseReviews"
  style={
    Object {
      "overflowX": "auto",
    }
  }
>
  <Loader
    active={false}
  />
  <div>
    <Connect(BackButton)
      preset="coursePage"
    />
    <Link
      style={
        Object {
          "textAlign": "center",
        }
      }
      to="/labtool/courses/TKT20010.2018.K.A.1"
    >
      <h2>
         
        Aineopintojen harjoitustyö: Tietorakenteet ja algoritmit (periodi IV, 2017-2018)
      </h2>
    </Link>
    <Card
      className="studentCard"
      color="yellow"
      fluid={true}
      key="0"
    >
      <CardContent>
        <Header
          as="h2"
        >
          Maarit Mirja
           
          Opiskelija
           (
          014578343
          )
          <HeaderSubheader>
            <a
              href="mailto:maarit.opiskelija@helsinki.invalid"
            >
              maarit.opiskelija@helsinki.invalid
            </a>
          </HeaderSubheader>
        </Header>
        <div
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          Tiran labraprojekti
          : 
          <RepoLink
            url="http://github.com/tiralabra1"
          />
          <br />
          <p
            className="noOther"
          >
            Has no other participation in this course
          </p>
        </div>
        <Button
          as="button"
          color="red"
          onClick={[Function]}
          style={
            Object {
              "float": "right",
            }
          }
        >
          Mark as dropped
        </Button>
      </CardContent>
    </Card>
    <span
      key="1"
    >
      <Button
        as="button"
        onClick={[Function]}
        size="small"
        type="button"
      >
        Show all reviews
      </Button>
      <br />
    </span>
    <Accordion
      fluid={true}
      id="review0"
      key="2"
      styled={true}
    >
      <AccordionTitle
        index={0}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 1
        , points 
        3
      </AccordionTitle>
      <AccordionContent>
        <h3>
          Review
        </h3>
        <Card
          color="yellow"
          fluid={true}
        >
          <CardContent>
            <h4>
               Points 
              3
               
            </h4>
            <h4>
               Feedback 
            </h4>
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={true}
              parserOptions={Object {}}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              sourcePos={false}
              transformLinkUri={[Function]}
            >
              Hienoa työtä!
            </ReactMarkdown>
             
            <span />
          </CardContent>
          <CardContent
            style={
              Object {
                "paddingBottom": "5px",
              }
            }
          >
            <Button
              as="button"
              onClick={[Function]}
              size="small"
              type="button"
            >
              Send email notification
            </Button>
          </CardContent>
        </Card>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/1"
        >
          <Popup
            content="Edit review"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
        <CommentGroup />
        <withRouter(Connect(LabtoolAddComment))
          allowHidden={true}
          commentFieldId="TKT20010.2018.K.A.1:10001"
          handleSubmit={[Function]}
          weekId={10001}
        />
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      id="review1"
      key="3"
      styled={true}
    >
      <AccordionTitle
        index={1}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 2
        , points 
        2
      </AccordionTitle>
      <AccordionContent>
        <h3>
          Review
        </h3>
        <Card
          color="yellow"
          fluid={true}
        >
          <CardContent>
            <h4>
               Points 
              2
               
            </h4>
            <h4>
               Feedback 
            </h4>
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={true}
              parserOptions={Object {}}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              sourcePos={false}
              transformLinkUri={[Function]}
            >
              Melko hienoa työtä!
            </ReactMarkdown>
             
            <div>
              <br />
              <h4>
                Review notes for instructors 
              </h4>
              <ReactMarkdown
                astPlugins={Array []}
                escapeHtml={true}
                parserOptions={Object {}}
                plugins={Array []}
                rawSourcePos={false}
                renderers={Object {}}
                skipHtml={false}
                sourcePos={false}
                transformLinkUri={[Function]}
              >
                READMEssa ongelmia
              </ReactMarkdown>
               
            </div>
          </CardContent>
          <CardContent
            style={
              Object {
                "paddingBottom": "5px",
              }
            }
          >
            <Button
              as="button"
              onClick={[Function]}
              size="small"
              type="button"
            >
              Send email notification
            </Button>
          </CardContent>
        </Card>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/2"
        >
          <Popup
            content="Edit review"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
        <CommentGroup />
        <withRouter(Connect(LabtoolAddComment))
          allowHidden={true}
          commentFieldId="TKT20010.2018.K.A.1:10002"
          handleSubmit={[Function]}
          weekId={10002}
        />
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      id="review2"
      key="4"
      styled={true}
    >
      <AccordionTitle
        index={2}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 3
        , points 
        3
      </AccordionTitle>
      <AccordionContent>
        <h3>
          Review
        </h3>
        <Card
          color="yellow"
          fluid={true}
        >
          <CardContent>
            <h4>
               Points 
              3
               
            </h4>
            <h4>
               Feedback 
            </h4>
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={true}
              parserOptions={Object {}}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              sourcePos={false}
              transformLinkUri={[Function]}
            >
              Erittäin hienoa työtä!
            </ReactMarkdown>
             
            <span />
          </CardContent>
          <CardContent
            style={
              Object {
                "paddingBottom": "5px",
              }
            }
          >
            <Button
              as="button"
              onClick={[Function]}
              size="small"
              type="button"
            >
              Send email notification
            </Button>
          </CardContent>
        </Card>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/3"
        >
          <Popup
            content="Edit review"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
        <CommentGroup />
        <withRouter(Connect(LabtoolAddComment))
          allowHidden={true}
          commentFieldId="TKT20010.2018.K.A.1:10003"
          handleSubmit={[Function]}
          weekId={10003}
        />
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      id="review3"
      key="5"
      styled={true}
    >
      <AccordionTitle
        index={3}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 4
        , points 
        3
      </AccordionTitle>
      <AccordionContent>
        <h3>
          Review
        </h3>
        <Card
          color="yellow"
          fluid={true}
        >
          <CardContent>
            <h4>
               Points 
              3
               
            </h4>
            <h4>
               Feedback 
            </h4>
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={true}
              parserOptions={Object {}}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              sourcePos={false}
              transformLinkUri={[Function]}
            >
              Hyvin menee!
            </ReactMarkdown>
             
            <span />
          </CardContent>
          <CardContent
            style={
              Object {
                "paddingBottom": "5px",
              }
            }
          >
            <Button
              as="button"
              onClick={[Function]}
              size="small"
              type="button"
            >
              Send email notification
            </Button>
          </CardContent>
        </Card>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/4"
        >
          <Popup
            content="Edit review"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
        <CommentGroup />
        <withRouter(Connect(LabtoolAddComment))
          allowHidden={true}
          commentFieldId="TKT20010.2018.K.A.1:10004"
          handleSubmit={[Function]}
          weekId={10004}
        />
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      id="review4"
      key="6"
      styled={true}
    >
      <AccordionTitle
        index={4}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 5
         
      </AccordionTitle>
      <AccordionContent>
        <h3>
          Review
        </h3>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/5"
        >
          <Popup
            content="Review week"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
        <h4>
           Not Graded 
        </h4>
        <h4>
           No comments 
        </h4>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      id="review5"
      key="7"
      styled={true}
    >
      <AccordionTitle
        index={5}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 6
         
      </AccordionTitle>
      <AccordionContent>
        <h3>
          Review
        </h3>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/6"
        >
          <Popup
            content="Review week"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
        <h4>
           Not Graded 
        </h4>
        <h4>
           No comments 
        </h4>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      id="review6"
      key="8"
      styled={true}
    >
      <AccordionTitle
        index={6}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         
        Week 7
         
      </AccordionTitle>
      <AccordionContent>
        <h3>
          Review
        </h3>
        <Link
          to="/labtool/reviewstudent/TKT20010.2018.K.A.1/10011/7"
        >
          <Popup
            content="Review week"
            disabled={false}
            eventsEnabled={true}
            offset={0}
            on={
              Array [
                "click",
                "hover",
              ]
            }
            pinned={false}
            position="top left"
            trigger={
              <Button
                as="button"
                circular={true}
                color="orange"
                icon={
                  Object {
                    "color": "black",
                    "name": "edit",
                    "size": "large",
                  }
                }
                size="tiny"
              />
            }
          />
        </Link>
        <h4>
           Not Graded 
        </h4>
        <h4>
           No comments 
        </h4>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      id="review7"
      key="9"
      styled={true}
    >
       
      <AccordionTitle
        index={8}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         Code Review 
         
      </AccordionTitle>
      <AccordionContent>
        <p>
          <strong>
            Project to review:
          </strong>
           
          Tira super projekti
           
          <br />
          <strong>
            GitHub:
          </strong>
           
          <a
            href="http://github.com/superprojekti"
            rel="noopener noreferrer"
            target="_blank"
          >
            http://github.com/superprojekti
          </a>
        </p>
        <strong>
          Code review:
        </strong>
         
        <a
          href="https:github.com/toReview/repo/pull/1"
          rel="noopener noreferrer"
          target="_blank"
        >
          https:github.com/toReview/repo/pull/1
        </a>
        <h4>
          Not graded yet
        </h4>
        <Form
          as="form"
          onSubmit={[Function]}
        >
          <label>
            Points 
          </label>
          <Input
            defaultValue=""
            name="points"
            step="0.01"
            style={
              Object {
                "width": "100px",
              }
            }
            type="number"
          />
          <Input
            type="submit"
            value="Grade"
          />
        </Form>
      </AccordionContent>
    </Accordion>
    <Accordion
      fluid={true}
      id="review8"
      key="10"
      styled={true}
    >
       
      <AccordionTitle
        index={9}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="dropdown"
        />
         Code Review 
         
      </AccordionTitle>
      <AccordionContent>
        <p>
          <span>
            This student is assigned a repo that doesn't belong to this course to review
          </span>
          <br />
          <strong>
            GitHub:
          </strong>
           
          <a
            href="https://github.com/userName/arbitraryRepo"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://github.com/userName/arbitraryRepo
          </a>
        </p>
        <strong>
          Code review:
        </strong>
         
        No review linked yet
        <h4>
          Not graded yet
        </h4>
        <Form
          as="form"
          onSubmit={[Function]}
        >
          <label>
            Points 
          </label>
          <Input
            defaultValue=""
            name="points"
            step="0.01"
            style={
              Object {
                "width": "100px",
              }
            }
            type="number"
          />
          <Input
            type="submit"
            value="Grade"
          />
        </Form>
      </AccordionContent>
    </Accordion>
  </div>
</div>
`;
